andela/ah-jawans-backend
# #166790090: view and amend profile

#### What does this PR do?
This PR has the feature Users should be able to create their profiles
- I have added Files in controllers and middleware to verify profile
- I have added tests for the same.
- I have added Swagger documentation.

#### Description of Task to be completed?
This has the following endpoint working -

GET      ``/api/user/:username``
PATCH ``/api/users/:username``

#### How should this be manually tested?
1. clone the repository with this branch (ft-create-profile-166790090)
2. go to your terminal and then write the following command
``cd ah-jawan-backend``
3. after that try to install all dependencies that are required for this project, simply by running the following command:
``npm install``
4. Then running the server, by running the below commands:
npm migrate for this command make sure that the sequelize-cli package is installed globally
``npm run dev``
5. After running the server, Open your postman, and then try to send for searching the article based on the provided tag, author's name, article's title or any keywords
for running the tests, write the below command in your terminal
``npm run test``

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/166790090

#### Screenshots (if appropriate)
<img width="876" alt="Screen Shot 2019-07-11 at 16 36 29" src="https://user-images.githubusercontent.com/51748364/61059948-2934fa00-a3fa-11e9-82da-aef76c48c6f3.png">
<img width="561" alt="Screen Shot 2019-07-11 at 16 37 10" src="https://user-images.githubusercontent.com/51748364/61059983-305c0800-a3fa-11e9-945b-dd8b61c0c7df.png">

#### Questions:
N/A
